src/gui/qgsrasterlayersaveasdialog.cpp

```diff
@@ -380,14 +401,55 @@ QString QgsRasterLayerSaveAsDialog::outputFileName() const
   return fileName;
 }
 
+QString QgsRasterLayerSaveAsDialog::outputLayerName() const
+{
+  if ( mLayerName->text().isEmpty() && outputFormat() == QStringLiteral( "GPKG" ) && !mTileModeCheckBox->isChecked() )
+  {
+    // Always return layer name for GeoPackages
+    return QFileInfo( mFilename->filePath() ).baseName();
+  }
+  else
+  {
+    return mLayerName->text();
+  }
+}
+
 QString QgsRasterLayerSaveAsDialog::outputFormat() const
 {
   return mFormatComboBox->currentData().toString();
 }
 
 QStringList QgsRasterLayerSaveAsDialog::createOptions() const
 {
-  return mCreateOptionsGroupBox->isChecked() ? mCreateOptionsWidget->options() : QStringList();
+  QStringList options = mCreateOptionsGroupBox->isChecked() ? mCreateOptionsWidget->options() : QStringList();
+  if ( outputFormat() == QStringLiteral( "GPKG" ) )
+  {
+    // Overwrite the GPKG table options
+    int indx = options.indexOf( QRegExp( "^RASTER_TABLE=.*" ) );
```

Comment:
Use QRegularExpression here instead of the older (likely to be removed in qt6!) QRegExp class

Reply:
Done, thanks